### ISSUE TITLE
Tools.IsFullPath returns false for linux paths

### ISSUE THREAD

**Issue by timhill-iress** [axuno/MailMergeLib#19]
While attempting to use MailMergeLib to send emails with dotnetcore on linux I get a failure.

'Building of message failed with one or more exceptions. See inner exceptions for details.'
 Inner exceptions found, see $exception in variables window for more details.
 Innermost exception 	 System.IO.DirectoryNotFoundException : 'Config.FileBaseDirectory' is not a full path.

It doesn't seem possible to work around this on linux as Tools.IsFullPath either returns false as the path isn't a valid looking linux path or because the root returned from Path.GetPathRoot(path) is only ever 1 character '/'

This is the code that seems too harsh for a linux path

if (pathRoot.Length <= 2) // Accepts X:\ and \\UNC\PATH, rejects empty string, \ and X:
                return false;

**Comment by axunonb** [comment]
Hi Tim, even without looking at the code, the explanation is I never used it in another environment than windows. So thanks. Would really appreciate a PR with Unit Test to run under Linux as well. Could you help me out here?

**Comment by timhill-iress** [comment]
Happy to help.

I think there might be a couple of challenges.
* The MailMergeLib project file needs a tweak just to get it to build on linux. I'm not sure of the best way to manage this, maybe a different build config?
* The Test Project seems windows/Net4.6 specific,  it might need a little more work to get it to build and run with dotnetcore on linux. Again not sure how to feed this back in..
* The current path tests are windows specific, I'd need a few linux specific tests. Do you have any thoughts on the best pattern for this. Simplest might be to  have a linux version of the test function and a windows version with their own test cases. Then the test functions use https://github.com/nunit/docs/wiki/Assumptions to check it's the appropriate OS. So something like  

    Assume.That(RuntimeInformation.IsOSPlatform(OSPlatform.Windows));

To check if we are on Windows.

Let me know what you think, and I'll try to help out.

**Comment by axunonb** [comment]
Earlier today I took some time for this issue:

* MailMergeLib project file tweaks for Linux: Not sure why this should be necessary, maybe could give me a clue.
* Test Project **was** Windows/Net46 specific until today. I updated the Unit Test project to now target netcoreapp2.0 and net461
* The path specific code should actually all be in the static ```MailMergeLib.Tools``` class. The platform depended code logic could go there. We would need to determine on which platform MailMergeLib is running. For that, we could go for [this suggestion](https://stackoverflow.com/questions/38790802/determine-operating-system-in-net-core). Then we wouldn't need separate Linux version of MailMergeLib.

Would this be a starting point now for you?

**Comment by timhill-iress** [comment]
Hi, just to give an update as I'd hoped to do a PR yesterday.
* No need to change the MailMergeLib project file after all. I just needed to pass the target in to the dotnet build command.
* Test project converted to netcoreapp2.0 is a great help
* Implemented the change as suggested.
* Using the ExcludePlatform / IncludePlatform feature of NUnit to make specific testcases Linux or Windows specific (https://github.com/nunit/docs/wiki/TestCase-Attribute)

It's just taking me a little longer as there is one final testcase I can't get running on Linux - CancelSendOperationWithDelay.

I'm sure it's nothing to do with my change. If you are happy it runs Ok on windows I could exclude it on linux.

**Comment by timhill-iress** [comment]
PR is from my private github account rather than my work account. @thegruffalo.

**Comment by axunonb** [comment]
Hi Tim, great news, thanks a lot. I'm currently on a business trip and I'll come back to you latest this weekend.

**Comment by axunonb** [comment]
Tim, I was too curious, and had an immediate glance: Looks good.
Actually the test case ```CancelSendOperationWithDelay``` not working with Linux is strange. Could you please elaborate? I mean: It's just a regular cancellation of an async task, which should not be platform dependent, should it?

**Comment by axunonb** [comment]
With the changes for Linux, [file system specifics of iOS/macOS](https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/Introduction/Introduction.html#//apple_ref/doc/uid/TP40010672-CH1-SW1) would also be covered. Do you agree?

**Comment by thegruffalo** [comment]
I had a look at the Asynchronous Cancel, and I couldn’t see why it wouldn’t work. The result I got was the sent count was always 1, so the email had been sent. I tried reducing the timeout in case maybe it was quite time/ environment sensitive but even with much lower times, say 10ms, it still failed. I can take a look again, perhaps this weekend.

The code change should cover MacOSX but the test changes may not, probably need to add in MacOSX to the include/ exclude platforms.

Could really do with someone with a Mac to validate that though.
